src/commands/createStaticWebApp/ApiLocationStep.ts

```diff
@@ -3,25 +3,31 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizardPromptStep } from "vscode-azureextensionui";
-import { apiSubpathSetting, defaultApiName } from "../../constants";
+import { AzureWizardPromptStep, IWizardOptions } from "vscode-azureextensionui";
+import { defaultApiLocation, enterInputQuickPickItem } from "../../constants";
 import { ext } from "../../extensionVariables";
+import { getGitTreeQuickPicks } from "../../utils/gitHubUtils";
 import { localize } from "../../utils/localize";
-import { getWorkspaceSetting } from "../../utils/settingsUtils";
 import { addLocationTelemetry } from "./addLocationTelemetry";
+import { EnterApiLocationStep } from "./EnterApiLocationStep";
 import { IStaticWebAppWizardContext } from "./IStaticWebAppWizardContext";
 
 export class ApiLocationStep extends AzureWizardPromptStep<IStaticWebAppWizardContext> {
     public async prompt(wizardContext: IStaticWebAppWizardContext): Promise<void> {
-        wizardContext.apiLocation = (await ext.ui.showInputBox({
-            value: getWorkspaceSetting(apiSubpathSetting, wizardContext.fsPath) || defaultApiName,
-            prompt: localize('apiLocation', "Enter the location of your Azure Functions code (Leave blank for no Azure Functions code)"),
-        })).trim();
-        addLocationTelemetry(wizardContext, 'apiLocation', defaultApiName);
+        const placeHolder: string = localize('apiLocation', "Select the location of your Azure Functions code");
+        (await ext.ui.showQuickPick(getGitTreeQuickPicks(wizardContext, true), { placeHolder, suppressPersistence: true })).data.trim();
+
+        wizardContext.manuallyEnterApi = wizardContext.apiLocation === enterInputQuickPickItem.data;
```

Comment:
Here's how I would do this:
- Remove `manuallyEnterApi` from the wizard context
- Set `data` to `undefined` in the "manually enter" case
- Below in `getSubWizard`, check `wizardContext.apiLocation === undefined`

It feels weird to have a "QuickPickItem" as a "constant" (normally the constants.ts file is just for strings/numbers). Plus technically this blocks people from using "manuallyEnter" as the location name (I know - why would they 😝). Finally, it just cleans up the wizardContext a bit